src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/DeliverySessionService.kt

```diff
@@ -109,7 +109,7 @@ class DeliverySessionService(
       appointmentTime = appointmentTime,
       durationInMinutes = durationInMinutes,
       referral = session.referral,
-      superseded = session.appointments.size >= 1
+      superseded = attended == null,
```

Comment:
all new appointments are created with attended = null, this is marking them as superseded which isn't correct

Reply:
Instead of these changes to `DeliverySessionService.kt` would it not be better to let `AppointmentService.kt` handle the superseded flag?

It has existing processing 
```
  private fun updateAppointment(
   ....
    oldAppointment.superseded = true
    appointmentRepository.save(oldAppointment)
   ...
```
which does exactly what we want, but this method isn't called in the current reschedule appointment processing - so either find which method is being used and update it to include the above, or refactor to use updateAppointment as part of the call.  What do you think @kth13 ?

Reply:
alternatively (as deliverysessionservice already has `// TODO: Some code duplication here with AppointmentService.kt`)
in `DeliverySessionService.kt` line 227 something like 
```diff
    appointmentRepository.saveAndFlush(appointment)
+    existingAppointment.superseded = true
+    appointmentRepository.save(existingAppointment)
```
(repeated anywhere else we are creating new appointments where necessary)

should do it, shouldn't it?

Reply:
After discussion with @kth13 this change would require another ticket because its quite a big change however superseded by default has been changed to only when rescheduled. see future commit.